addons/math-in-inputs/userscript.js

```diff
@@ -0,0 +1,163 @@
+import { shiftDecimalPointToRight, shiftDecimalPointToLeft } from "../../libraries/common/cs/math-on-decimals.js";
+
+export default async function ({ addon }) {
+  function getRegexFromSettings() {
+    let textInInputs = addon.settings.get("textInInputs");
+    return textInInputs ? /^.*$/i : /^[0-9+\-*^/(). ]+$/;
+  }
+  const checkElForTypeChange = (el, focus) => {
```

Comment:
Instead of using lots of if/elses for each condition to check if to return true or false. Here's some cleaner code you could use:

```js
 const checkElForTypeChange = (el, focus) => {
    if (!el.classList) return false;
    const type = focus === "in" ? "input[type=number]" : "input[type=text]";
    const selectors = [
      "[class*=mediaRecorderPopupContent]",
      "[class*=input_input-form_]",
      "[class*=Popover-body]",
      "[class*=sa-paint-snap-settings]",
      "[class*=sa-onion-settings]",
    ];
    return selectors.some(selector => el.matches(`${selector}${type}`));
  };```

Reply:
I haven't tested it but from my reasoning of it, this wont work. When we match to "[class*=input_input-form_]" this in itself doesnt return true, but then allows more specific matches to continue. If we don't do this, I *think* it will start allowing math in inputs that we dont want it to, which is how you have implemented it. Currently if any input matches "[class*=input_input-form_]" in your suggested changes, it will immediately return true, which we do NOT want, because there are some input that match this that we do not want to do math in.

Reply:
For example, the costume name input (unless we somehow decide that it should do math in ALL inputs no matter how...odd it might seem?)
![image](https://github.com/ScratchAddons/ScratchAddons/assets/81956724/3569f677-536f-4483-b2d0-606954e374c7)